doc/release/upcoming_changes/26081.improvement.rst

```diff
@@ -0,0 +1,79 @@
+``ndarray`` Shapes are now covariant and bound
+---------------------------------------------------------------
+Static typing for ``ndarray`` is a long-term effort that continues 
+with this change.  Arrays are a generic type, parameterized by both 
+shape and data type.  The shape parameter could previously be assigned
```

Comment:
```suggestion
shape and data type.  The previous shape-type placeholder could previously be assigned
```
Would emphasis here that previously we were truly dealing with a placeholder type slot.